inference-engine/tests/functional/plugin/conformance/subgraphs_dumper/src/matchers/single_op.cpp

```diff
@@ -0,0 +1,166 @@
+        case ngraph::element::Type_t::i32:
+            return compare_constants_data<int32_t>(op, ref);
+        case ngraph::element::Type_t::i64:
+            return compare_constants_data<int64_t>(op, ref);
+            // TODO cast_vector doesn't support u1 now
+//        case ngraph::element::Type_t::u1:
+//            return compare_constants_data<char>(op, ref);
+        case ngraph::element::Type_t::u8:
+            return compare_constants_data<uint8_t>(op, ref);
+        case ngraph::element::Type_t::u16:
+            return compare_constants_data<uint16_t>(op, ref);
+        case ngraph::element::Type_t::u32:
+            return compare_constants_data<uint32_t>(op, ref);
+        case ngraph::element::Type_t::u64:
+            return compare_constants_data<uint64_t>(op, ref);
+        default:
+            std::cerr << "Can't compare constants" << op << " with " << ref << "\n" << "Unsupported data type";
+            return false;
+    }
+}
+
+const char *SingleOpMatcher::name = "generic_single_op";
+bool
+SingleOpMatcher::same_op_type(const std::shared_ptr<ngraph::Node> &node, const std::shared_ptr<ngraph::Node> &ref) {
+    return node->get_type_info().name == ref->get_type_info().name &&
+           node->get_type_info().version == ref->get_type_info().version;
+}
+
+bool
+SingleOpMatcher::match_inputs(const std::shared_ptr<ngraph::Node> &node, const std::shared_ptr<ngraph::Node> &ref) {
+    if (node->get_input_size() != ref->get_input_size()) {
+        return false;
+    } else {
+        for (size_t i = 0; i < node->get_input_size(); ++i) {
+            bool rankIsEqual = node->get_input_tensor(i).get_partial_shape().rank() ==
+                               ref->get_input_tensor(i).get_partial_shape().rank();
+            bool elemTypeIsEqual = node->get_input_tensor(i).get_element_type() ==
+                                   ref->get_input_tensor(i).get_element_type();
+            bool is_dynamic = node->get_input_node_ptr(i)->is_dynamic() ==
+                              ref->get_input_node_ptr(i)->is_dynamic();
+            if (!(rankIsEqual && elemTypeIsEqual && is_dynamic)) {
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+bool
+SingleOpMatcher::match_outputs(const std::shared_ptr<ngraph::Node> &node, const std::shared_ptr<ngraph::Node> &ref) {
+    if (node->get_output_size() != ref->get_output_size()) {
+        return false;
+    } else {
+        // Match output element type
+        for (size_t i = 0; i < node->get_output_size(); ++i) {
+            if (node->get_output_tensor(i).get_element_type() !=
+                ref->get_output_tensor(i).get_element_type()) {
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+bool SingleOpMatcher::same_attrs(const std::shared_ptr<ngraph::Node> &node, const std::shared_ptr<ngraph::Node> &ref) {
+    return attributes::compare(node.get(), ref.get(), Comparator::CmpValues::ATTRIBUTES).valid;
```

Comment:
Will check whole subgraph if node is TI or Loop. Is this expected?

Reply:
Well, it's a subject to discuss which approach for tensor iterators and loops extracting we should implement. 
For now I guess it can be kept as is with whole body check, but in future we will need separate Matcher implementation to work with Tis and Loops properly.